packages/envd/internal/host/metrics.go

```diff
@@ -0,0 +1,39 @@
+package host
+
+import (
+	"math"
+	"time"
+
+	"github.com/shirou/gopsutil/v4/cpu"
+	"github.com/shirou/gopsutil/v4/mem"
+)
+
+type Metrics struct {
+	CPU       float64 `json:"cpu_pct"`   // Percent rounded to 2 decimal places
+	Mem       uint64  `json:"mem_bytes"` // Total virtual memory in bytes
+	Timestamp int64   `json:"ts"`        // Unix Timestamp in UTC
+}
+
+func GetMetrics() (*Metrics, error) {
+	v, err := mem.VirtualMemory()
+	if err != nil {
+		return nil, err
+	}
+
+	cpuPcts, err := cpu.Percent(time.Second, false)
```

Comment:
This will be blocking for 1 second, not sure if this make sense for the last call when we kill the sandbox. We know we want to collect these information, I am thinking if we don't want to run something in background and you would only pickup the current value or something like that?

Reply:
per `gopsutil` codebase:
```go
// Percent calculates the percentage of cpu used either per CPU or combined.
// If an interval of 0 is given it will compare the current cpu times against the last call.
// Returns one value per cpu, or a single value if percpu is set to false.
func Percent(interval time.Duration, percpu bool) ([]float64, error) {
```

I set it to `0` and it's much snappier now